data/summary/types/cgm_test.go

```diff
@@ -94,6 +96,36 @@ func NewDataSetCGMDataRanges(startTime time.Time, hours float64, ranges DataRang
 	return dataSetData
 }
 
+func NewDataSetCGMVariance(startTime time.Time, hours int, perHour int, SD float64) ([]data.Datum, float64) {
+	requiredRecords := hours * perHour
+	typ := pointer.FromString("cbg")
+
+	var dataSetData = make([]data.Datum, requiredRecords)
+	var uploadId = test.RandomSetID()
+	var deviceId = "SummaryTestDevice"
+
+	// dedicated rand, to be sure the sequence is kept just in case.
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	var values []float64
+
+	// generate requiredRecords of data
+	for count := 0; count < requiredRecords; count += perHour {
+		for inHour := 0; inHour < perHour; inHour++ {
+			datumTime := startTime.Add(time.Duration(-(count + inHour + 1)) * time.Hour / time.Duration(perHour))
+
+			datum := NewGlucose(typ, pointer.FromString(units), &datumTime, &deviceId, &uploadId)
```

Comment:
super nit: you could pull this out of the loops for a small speed bump. :)

Reply:
Maybe its unclear as github highlights a section and im reading this wrong, but how would we avoid NewGlucose in the loop? Reuse with a copy might be faster, but also a bit less clear, the unit test already runs near instantly.

Reply:
I did move all the repeated pointer functions out of the loop, perhaps thats what was meant.

Reply:
Oh yeah, sorry, it was just the `pointer.FromString(units)` that I was trying to highlight. I probably fat-fingered the selection when I was commenting and didn't realize.